src/webserver2/RPCExecutor.h

```diff
@@ -0,0 +1,210 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of clio: https://github.com/XRPLF/clio
+    Copyright (c) 2023, the clio developers.
+
+    Permission to use, copy, modify, and distribute this software for any
+    purpose with or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL,  DIRECT,  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#pragma once
+
+#include <rpc/Factories.h>
+#include <rpc/RPCHelpers.h>
+#include <util/Profiler.h>
+
+#include <boost/json/parse.hpp>
+
+/**
+ * @brief The executor for RPC requests called by web server
+ */
+template <class Engine, class ETL>
+class RPCExecutor
+{
+    std::shared_ptr<BackendInterface const> const backend_;
+    std::shared_ptr<Engine> const rpcEngine_;
+    std::shared_ptr<ETL const> const etl_;
+    // subscription manager holds the shared_ptr of this class
+    std::weak_ptr<SubscriptionManager> const subscriptions_;
+    util::TagDecoratorFactory const tagFactory_;
+
+    clio::Logger log{"RPC"};
```

Comment:
This class is extracted from the old handlerRequest. But I change the log  to "RPC", the logic happening here is more RPC related.